src/verilog/verilog_expr.h

```diff
@@ -1132,4 +1132,59 @@ extern inline verilog_assumet &to_verilog_assume(exprt &expr)
   return static_cast<verilog_assumet &>(expr);
 }
 
+class verilog_module_sourcet : public irept
+{
+public:
+  irep_idt name() const
+  {
+    return get(ID_name);
+  }
```

Comment:
Should there be some constructor that sets field(s)? Perhaps at least the name?

Reply:
Will come -- I'll replace `verilog_modulet` by this class.